tests/unit/test_power_grid_model.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from copy import copy
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from power_grid_model import PowerGridModel, initialize_array
+from power_grid_model.errors import PowerGridBatchError, PowerGridError
+
+from .utils import compare_result, import_case_data
+
+"""
+Testing network
+
+source_1(1.0 p.u., 100.0 V) --internal_impedance(j10.0 ohm, sk=1000.0 VA, rx_ratio=0.0)--
+-- node_0 (100.0 V) --load_2(const_i, -j5.0A, 0.0 W, 500.0 var)
+
+u0 = 100.0 V - (j10.0 ohm * -j5.0 A) = 50.0 V
+
+update_0:
+    u_ref = 0.5 p.u. (50.0 V)
+    q_specified = 100 var (-j1.0A)
+u0 = 50.0 V - (j10.0 ohm * -j1.0 A) = 40.0 V
+
+update_1:
+    q_specified = 300 var (-j3.0A)
+u0 = 100.0 V - (j10.0 ohm * -j3.0 A) = 70.0 V
+"""
+
+
+@pytest.fixture
+def case_data():
+    case_dir = Path(__file__).parent.parent / "data" / "unit_test"
+    return import_case_data(case_dir, sym=True)
```

Comment:
I think for unit tests it would be better to include the data in the test file instead of separate .json files.